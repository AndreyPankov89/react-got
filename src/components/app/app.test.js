import React from 'react';
import App from './app';
import {shallow} from 'enzyme';

describe('Testing <App/>', () => {
    const item = shallow(<App/>);
    describe('testing shap & state', () =>{
        it('App have rendered correctly', () => {
            expect(item).toMatchSnapshot();
        });

        it("App state error is false", () =>{
            expect(item.state().error).toBeFalsy();
        })

        it("App state visible is false", () =>{
            expect(item.state().visible).toBeFalsy();
        })

        it("App state selectedChar is null", () =>{
            expect(item.state().selectedChar).toBeNull();
        })

        it("App state text is 'on'", () =>{
            expect(item.state().visibleButtonText).toEqual('on');
        })
    });
    describe('Handlers tests', () => {
        it('testing onCharSelected', () => {
            item.instance().onCharSelected
            (1);
            expect(item.state().selectedChar).toEqual(1)
        });
        it('testing toggleVisible', () => {
            item.instance().toggleVisible();
            expect(item.state().visible).toBeTruthy();
            expect(item.state().visibleButtonText).toEqual('off');
        });
        
    })
})