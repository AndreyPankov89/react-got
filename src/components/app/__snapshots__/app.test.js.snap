// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> testing shap & state App have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
      >
        <Container
          tag="div"
        >
          <Header
            onToggleVisible={[Function]}
            visibleButtonText="on"
          />
        </Container>
        <Container
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              lg={
                Object {
                  "offset": 0,
                  "size": 5,
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            />
          </Row>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/characters"
            />
            <Route
              component={[Function]}
              path="/houses"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/books"
            />
            <Route
              path="/books/:id"
              render={[Function]}
            />
            <Route
              render={[Function]}
            />
          </Switch>
        </Container>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Container
            tag="div"
          >
            <Header
              onToggleVisible={[Function]}
              visibleButtonText="on"
            />
          </Container>,
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              />
            </Row>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/characters"
              />
              <Route
                component={[Function]}
                path="/houses"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/books"
              />
              <Route
                path="/books/:id"
                render={[Function]}
              />
              <Route
                render={[Function]}
              />
            </Switch>
          </Container>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Header
              onToggleVisible={[Function]}
              visibleButtonText="on"
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onToggleVisible": [Function],
              "visibleButtonText": "on",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />
              </Row>,
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  path="/characters"
                />
                <Route
                  component={[Function]}
                  path="/houses"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/books"
                />
                <Route
                  path="/books/:id"
                  render={[Function]}
                />
                <Route
                  render={[Function]}
                />
              </Switch>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": null,
                  "lg": Object {
                    "offset": 0,
                    "size": 5,
                  },
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Route
                    component={[Function]}
                    path="/characters"
                  />,
                  <Route
                    component={[Function]}
                    path="/houses"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/books"
                  />,
                  <Route
                    path="/books/:id"
                    render={[Function]}
                  />,
                  <Route
                    render={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/characters",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/houses",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/books",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/books/:id",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
        >
          <Container
            tag="div"
          >
            <Header
              onToggleVisible={[Function]}
              visibleButtonText="on"
            />
          </Container>
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              />
            </Row>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/characters"
              />
              <Route
                component={[Function]}
                path="/houses"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/books"
              />
              <Route
                path="/books/:id"
                render={[Function]}
              />
              <Route
                render={[Function]}
              />
            </Switch>
          </Container>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Container
              tag="div"
            >
              <Header
                onToggleVisible={[Function]}
                visibleButtonText="on"
              />
            </Container>,
            <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />
              </Row>
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  path="/characters"
                />
                <Route
                  component={[Function]}
                  path="/houses"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/books"
                />
                <Route
                  path="/books/:id"
                  render={[Function]}
                />
                <Route
                  render={[Function]}
                />
              </Switch>
            </Container>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Header
                onToggleVisible={[Function]}
                visibleButtonText="on"
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onToggleVisible": [Function],
                "visibleButtonText": "on",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Row
                  tag="div"
                >
                  <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  />
                </Row>,
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    path="/characters"
                  />
                  <Route
                    component={[Function]}
                    path="/houses"
                  />
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/books"
                  />
                  <Route
                    path="/books/:id"
                    render={[Function]}
                  />
                  <Route
                    render={[Function]}
                  />
                </Switch>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  />,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "lg": Object {
                      "offset": 0,
                      "size": 5,
                    },
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />,
                    <Route
                      component={[Function]}
                      path="/characters"
                    />,
                    <Route
                      component={[Function]}
                      path="/houses"
                    />,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/books"
                    />,
                    <Route
                      path="/books/:id"
                      render={[Function]}
                    />,
                    <Route
                      render={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/characters",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/houses",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/books",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "path": "/books/:id",
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
